ess/src/API/EMBC.ESS/Managers/Admin/AdminManager.cs

```diff
@@ -223,5 +223,54 @@ public async Task<string> Handle(DeactivateSupplierCommand cmd)
 
             return res.SupplierId;
         }
+
+        public async Task<string> Handle(ClaimSupplierCommand cmd)
+        {
+            var supplier = (await supplierRepository.QuerySupplier(new SupplierSearchQuery
+            {
+                SupplierId = cmd.SupplierId,
+            })).Items.SingleOrDefault(m => m.Id == cmd.SupplierId);
+            if (supplier == null) throw new NotFoundException($"Supplier {cmd.SupplierId} not found", cmd.SupplierId);
+            if (supplier.Team != null && supplier.Team.Id != null) throw new Exception($"Supplier already has a primary team");
+
+            supplier.Team = new Resources.Suppliers.Team { Id = cmd.TeamId };
+            supplier.Status = Resources.Suppliers.SupplierStatus.Active;
```

Comment:
this may be dangerous to assume a supplier will be activated. There may be a reason that it is not active

Reply:
According to the user story, when a supplier is claimed it is supposed to default to Active.